conn.go

```diff
@@ -193,7 +194,9 @@ func (l *Conn) Close() {
 // SetTimeout sets the time after a request is sent that a MessageTimeout triggers
 func (l *Conn) SetTimeout(timeout time.Duration) {
 	if timeout > 0 {
+		l.timeoutMutex.Lock()
 		l.requestTimeout = timeout
+		l.timeoutMutex.Unlock()
```

Comment:
Might be simpler to make this an atomic value and do lockless load/store. We already have a shim for < go1.3 using locks